src/Microsoft.DotNet.Configurer/IFileExtensions.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.Configurer
+{
+    internal static class FileSystemExtensions
+    {
+        public static void CreateIfNotExists(this IFileSystem fileSystem, string filePath)
```

Comment:
Why an extension method vs implementing this on the interface?  The problem with an extension method is that if a unit test is calling this and expects a mock interaction, it won't receive one.

Reply:
Actually the type is on IFileSystem, so if the impl is a mock it will still use the mock.

At the same time. This class use both Util.dll and platform abstraction while they don't have dependency on each other.

Reply:
Good point.  I guess this is fine, provided whatever file system interactions the extensions methods do are limited to using the `IFileSystem`.